include/ucl.h

```diff
@@ -962,8 +970,9 @@ UCL_EXTERN void ucl_parser_register_macro (struct ucl_parser *parser,
  * @param macro macro name (without leading dot)
  * @param handler handler (it is called immediately after macro is parsed)
  * @param ud opaque user data for a handler
+ * @return 0 on success, -1 on failure (i.e. ENOMEM)
  */
-UCL_EXTERN void ucl_parser_register_context_macro (struct ucl_parser *parser,
+UCL_EXTERN int ucl_parser_register_context_macro (struct ucl_parser *parser,
```

Comment:
And here...

Reply:
> And here...

Done.

> You can get error by `ucl_parser_get_error` btw.

Hmm, I don't see how.  ucl_parser_register_context_macro() is not doing anything to register error with parser object.